src/Package/Impl/Repl/VsRInteractiveWorkflowProvider.cs

```diff
@@ -74,10 +79,20 @@ internal class VsRInteractiveWorkflowProvider : IRInteractiveWorkflowProvider, I
                                                     _plotsProvider, _activeTextViewTracker, _debuggerModeTracker, 
                                                     _shell, settings, _wss, () => DisposeInstance(sessionProvider));
             _disposableBag.Add(workflow);
+
+            sessionProvider.BrokerChanging += OnBrokerChanging;
             return workflow;
         }
 
+        private void OnBrokerChanging(object sender, EventArgs e) {
```

Comment:
I think there is a side effect that focus will be changed each time user changes a broker.
You can use `workflow.ActiveWindow.Container.Show(focus: false, immediate: false);` and move this handler inside workflow